Fortified string checks report out-of-bounds write

```
In function ‘memcpy’,
    inlined from ‘mdb_node_add’ at lmdb/mdb.c:8184:4:
/usr/include/x86_64-linux-gnu/bits/string_fortified.h:34:10: warning: writing 8 bytes into a region of size 1 [-Wstringop-overflow=]
   34 |   return __builtin___memcpy_chk (__dest, __src, __len, __bos0 (__dest));
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In function ‘memcpy’,
    inlined from ‘mdb_node_add’ at lmdb/mdb.c:8178:5:
/usr/include/x86_64-linux-gnu/bits/string_fortified.h:34:10: warning: writing 8 bytes into a region of size 1 [-Wstringop-overflow=]
   34 |   return __builtin___memcpy_chk (__dest, __src, __len, __bos0 (__dest));
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Requires very new gcc-10

--- a/.travis.yml
+++ b/.travis.yml
@@ -6,9 +6,9 @@ matrix:
   include:
     - os: linux
       env: R_CODECOV=true
-    - os: osx
-      osx_image: xcode8.3
-      latex: false
+#    - os: osx
+#      osx_image: xcode8.3
+#      latex: false
 
 r_packages:
   - covr

--- a/DESCRIPTION
+++ b/DESCRIPTION
@@ -1,6 +1,6 @@
 Package: thor
 Title: Interface to 'LMDB'
-Version: 1.1.1
+Version: 1.1.2
 Authors@R: c(person("Rich", "FitzJohn", role = c("aut", "cre"),
     email = "rich.fitzjohn@gmail.com"),
     person("Howard", "Chu", role = c("aut", "cph")),

--- a/DESCRIPTION
+++ b/DESCRIPTION
@@ -25,7 +25,7 @@ Suggests:
     knitr,
     rmarkdown,
     testthat
-RoxygenNote: 6.1.1
+RoxygenNote: 7.0.2
 VignetteBuilder: knitr
 ByteCompile: true
 Encoding: UTF-8

--- a/R/mdb_env.R
+++ b/R/mdb_env.R
@@ -131,10 +131,14 @@
 ##'   environment.
 ##'
 ##' @param mapsize Maximum size database may grow to; used to size the
-##'   memory mapping.  If database grows larger than ``map_size``, an
-##'   error will be thrown and the user must close and reopen the
-##'   \code{mdb_env}.  On 64-bit there is no penalty for making this
-##'   huge (say 1TB). Must be <2GB on 32-bit.
+##'   memory mapping. This is measured in bytes, and the default (as
+##'   set in lmdb) is only 1MB (2^20 bytes). If database grows larger
+##'   than \code{map_size}, an error will be thrown and the user must
+##'   close and reopen the \code{mdb_env}.  On 64-bit there is no
+##'   penalty for making this huge (say 1TB). Must be <2GB on 32-bit.
+##'   Increasing this may cause your operating system to report the
+##'   disk as being used while your database is open, though this is
+##'   just the amount reserved.
 ##'
 ##' @param reversekey Passed through to \code{open_database} for the
 ##'   main database.  If \code{TRUE}, keys are strings to be compared

--- a/man/mdb_env.Rd
+++ b/man/mdb_env.Rd
@@ -4,11 +4,24 @@
 \alias{mdb_env}
 \title{Create an mdb_env environment}
 \usage{
-mdb_env(path, mode = as.octmode("644"), subdir = TRUE,
-  readonly = FALSE, metasync = TRUE, sync = TRUE, writemap = FALSE,
-  lock = TRUE, mapasync = FALSE, rdahead = TRUE, meminit = TRUE,
-  maxdbs = NULL, maxreaders = NULL, mapsize = NULL,
-  reversekey = FALSE, create = TRUE)
+mdb_env(
+  path,
+  mode = as.octmode("644"),
+  subdir = TRUE,
+  readonly = FALSE,
+  metasync = TRUE,
+  sync = TRUE,
+  writemap = FALSE,
+  lock = TRUE,
+  mapasync = FALSE,
+  rdahead = TRUE,
+  meminit = TRUE,
+  maxdbs = NULL,
+  maxreaders = NULL,
+  mapsize = NULL,
+  reversekey = FALSE,
+  create = TRUE
+)
 }
 \arguments{
 \item{path}{The directory in which the database files will reside.
@@ -127,10 +140,14 @@ transactions.  Can only be set in the first process to open an
 environment.}
 
 \item{mapsize}{Maximum size database may grow to; used to size the
-memory mapping.  If database grows larger than ``map_size``, an
-error will be thrown and the user must close and reopen the
-\code{mdb_env}.  On 64-bit there is no penalty for making this
-huge (say 1TB). Must be <2GB on 32-bit.}
+memory mapping. This is measured in bytes, and the default (as
+set in lmdb) is only 1MB (2^20 bytes). If database grows larger
+than \code{map_size}, an error will be thrown and the user must
+close and reopen the \code{mdb_env}.  On 64-bit there is no
+penalty for making this huge (say 1TB). Must be <2GB on 32-bit.
+Increasing this may cause your operating system to report the
+disk as being used while your database is open, though this is
+just the amount reserved.}
 
 \item{reversekey}{Passed through to \code{open_database} for the
 main database.  If \code{TRUE}, keys are strings to be compared

--- a/man/storr_thor.Rd
+++ b/man/storr_thor.Rd
@@ -5,8 +5,12 @@
 \alias{driver_thor}
 \title{Thor driver for storr}
 \usage{
-storr_thor(env, prefix = "", hash_algorithm = NULL,
-  default_namespace = "objects")
+storr_thor(
+  env,
+  prefix = "",
+  hash_algorithm = NULL,
+  default_namespace = "objects"
+)
 
 driver_thor(env, prefix = "", hash_algorithm = NULL)
 }

--- a/src/lmdb/mdb.c
+++ b/src/lmdb/mdb.c
@@ -1075,7 +1075,7 @@ typedef struct MDB_node {
 /** @} */
 	unsigned short	mn_flags;		/**< @ref mdb_node */
 	unsigned short	mn_ksize;		/**< key size */
-	char		mn_data[1];			/**< key and data are appended here */
+	char		mn_data[sizeof(pgno_t)];			/**< key and data are appended here */
 } MDB_node;
 
 	/** Size of the node header, excluding dynamic data at the end */

--- a/src/lmdb/mdb.patch
+++ b/src/lmdb/mdb.patch
@@ -1,5 +1,16 @@
 diff --git a/src/lmdb/mdb.c b/src/lmdb/mdb.c
-index f5f99a5..309c0b5 100644
+index f7592bc..3554f1f 100644
+--- a/src/lmdb/mdb.c
++++ b/src/lmdb/mdb.c
+@@ -1075,7 +1075,7 @@ typedef struct MDB_node {
+ /** @} */
+ 	unsigned short	mn_flags;		/**< @ref mdb_node */
+ 	unsigned short	mn_ksize;		/**< key size */
+-	char		mn_data[1];			/**< key and data are appended here */
++	char		mn_data[sizeof(pgno_t)];			/**< key and data are appended here */
+ } MDB_node;
+ 
+ 	/** Size of the node header, excluding dynamic data at the end */
 --- a/src/lmdb/mdb.c
 +++ b/src/lmdb/mdb.c
 @@ -1733,7 +1733,7 @@ mdb_strerror(int err)

